test/ui_test_label_studio.py

```diff
@@ -0,0 +1,213 @@
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.firefox.firefox_binary import FirefoxBinary
+from selenium.common.exceptions import NoSuchElementException
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.action_chains import ActionChains
+
+
+class myTestCase(unittest.TestCase):
+    def setUp(self):
+        # self.driver = webdriver.Firefox()
+        #
+        self.downloadDir = './'
+        fp = webdriver.FirefoxProfile()
+        fp.set_preference("browser.download.folderList", 2)
+        fp.set_preference("browser.download.manager.showWhenStarting", False)
+        fp.set_preference("browser.download.dir", self.downloadDir)
+        # fp.set_preference("browser.helperApps.neverAsk.saveToDisk", "attachment/csv")
+        # fp.set_preference("browser.helperApps.neverAsk.saveToDisk", "attachment/json")
+        fp.set_preference("browser.helperApps.neverAsk.saveToDisk", "attachment/zip")
+        self.driver = webdriver.Firefox(firefox_profile=fp)
+        #
+        self.wait = WebDriverWait(self.driver, 10)
+        self.probel5 = '      '
+        #
+        self.site_url = 'http://localhost:8080/welcome'
+        #
+        self.site_logo = '//a[@id="logo"]/img'
+        self.welcome_Setup  = '//a[contains(text(), "Setup")]'
+        self.welcome_Import = '//a[contains(text(), "Import")]'
+        self.welcome_Start  = '//a[contains(text(), "Start")]'
+        self.welcome_Export = '//a[contains(text(), "Export")]'
+        #
+        #self.Setup_label_car = '//span[contains(text(), "Car")]'
+        #self.Setup_label_car = '//span[@class="ant-tag"]'
+        #self.Setup_label_car = '//span[@class="ant-tag"][contains(text(), "Car")]'
+        self.Setup_label_car = '/html/body/div/div/div/div/div/div/div[1]/div/div/div[2]/span[2]'
+        #
+        self.import_img_draganddrop_id = 'file-input'
+        # self.import_img_draganddrop_text = '//p[contains(text(), "Drag and drop your files here or click for import")]'
+        self.import_img_src = 'img/imgForDragAndDrop.jpg'
+        # self.import_img_src = '/Users/ahovhannes/Desktop/Hovo/h/JOB/Job_Label_Studio/label_studio_project/label-studio/test/img/imgForDragAndDrop.jpg'
+        #
+        # self.start_label_car = '//span[contains(text(), "Car")]'
+        self.start_label_car = 'ant-tag'
+        self.start_submit_btn = '//button/span[contains(text(), "Submit ")]'
+        #
+        self.export_submit_btn = '//a[contains(text(), "Export Completions")]'
+        #
+
+    def tearDown(self):
+        self.driver.close()
+
+    def test_cases(self):
+        print("..... Opening the site "+self.site_url+" .....")
+        self.driver.get(self.site_url)
+        print("..... Executing Test-Cases .....\n")
+        #
+        # self.step1('Setup your label config')
+        # self.goToWelcomePage()
```

Comment:
why is this step skipped? there would not be any config on the run from scratch, as I understand

Reply:
There were problem with iframe element finding process via selenium. After discussion with Nikolay, we decide to leave Setup step test implementation, because that page's iframe element can be changed to div element...